### PR TITLE
Add instructions to host the tutorial project on Glitch.com

### PR DESCRIPTION
**Pull request by bcopy**
Changes in this pull request:

- Provide cloud IDE setup instructions for Glitch.com (a user-friendly free web hosting solution with an integrated web IDE and command line access)
- Add instructions on how to protect the Django secret inside a .env file.

### CODE REVIEW

**Comment by bcopy** [comment]
Hello,

We will be relying on these updates for our upcoming public event at CERN

https://home.cern/news/announcement/cern/registration-open-online-programming-workshop-django-girls

and we would like to translate it also to French, and have it ready for instruction on 8th April to train our 25 coaches.
Can someone review, comment, merge as required ?

**Comment by das-g** [comment]
Hi @bcopy

Cool that you're doing an online Django Girls workshop! (I have myself been involved as coach an later as co-organizer in past Django Girls workshops in Switzerland, though all of them in the German-speaking parts of it. :wave:)

> and have it ready for instruction on 8th April

That's two week from now. While I hope we can make it until then, please be prepared for the case we won't manage. (The tutorial currently has only few still-active maintainers and all of them work on it only in their free time.) 

In case the change won't go through till then, you could generate the rendered version of your branch with [Honkit](https://honkit.netlify.app/) and host it yourself (or on GitHub pages or something similar). The most simple approach for that is probably to activate GitHub pages for your fork and adapt the condition https://github.com/DjangoGirls/tutorial/blob/57bba09425055903d95ed10472b57f10d06cd525/.github/workflows/deploy.yml#L10 in [the existing GitHub action config](https://github.com/DjangoGirls/tutorial/tree/master/.github/workflows).

**Comment by das-g** [comment]
Concerning putting the Django secret key into a `.env` file not tracked in Git: That's IMO a good idea, but can we somehow implement that independent of the development environment used, i.e., not only for Glitch.com?

**Comment by das-g** [comment]
Past and current issues and PRs regarding `SECRET_KEY`:
- #802
- #1192
- #1215
- DjangoGirls/tutorial-extensions#101
- #1687

It looks like the consensus until now was, that having the blatant (but probably not too serious) security issue of publishing the `SECRET_KEY` (and that of using the same one for development and production deployment) is more acceptable than rendering the Django Girls tutorial proper more complex, and that instead the actual solution should be presented in a (yet to be written) [tutorial extension](https://tutorial-extensions.djangogirls.org/).

@bcopy Does Glitch.com require that this be handled properly right from the beginning? Can the approach you take be generalized to work with all presented development environment setups, including local development?

**Comment by bcopy** [comment]
Dear Raphael,

Thank you for looking into our PR.

Regarding `SECRET_KEY` the Django documentation is adamant that it needs to be secured. However, this is left entirely to the end user. There are many tutorials on how to secure Django, but as far as I could see, the Django framework team do not have any open issues concerning a potential securing of the `settings.py ` parameters. So they consider that if you version control your Django project, you must edit your `ignore` configuration accordingly.

https://code.djangoproject.com/query?status=closed&status=new&description=~SECRET_KEY&desc=1&order=id

I agree that the tutorial should remain simple and securing concerns can certainly go in an extension.

> Does Glitch.com require that this be handled properly right from the beginning? 
> Can the approach you take be generalized to work with all presented development 
> environment setups, including local development?

Yes, in the case of Glitch, the `settings.py` file needs to be updated immediately, as indicated, or the Django website will not work at all. The SECRET_KEY could be left until later, but while we're at it, it makes sense to take action immediately.

Thank you for your tips concerning regenerating the documentation into our Github Pages, I will apply it.

I would like to contribute French translations via Crowdin though - but maybe you also have a suggestion on how we can make some progress without waiting for a merge to master.

File: .github/workflows/deploy.yml
Comment (das-g) [inline .github/workflows/deploy.yml:10]: Oops, sorry. I forgot that any changes to the branch would also be included in the Pull Request. Obviously, we can't merge that change or the action would break for this upstream repo here ([djangogirls/tutorial](https://github.com/DjangoGirls/tutorial)).

**Review by das-g — CHANGES_REQUESTED** [review]

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:13]:
```suggestion
Here is the instruction for cloud IDEs (PaizaCloud Cloud IDE, AWS Cloud9, Glitch.com).
```
or
```suggestion
Here are the instructions for cloud IDEs (PaizaCloud Cloud IDE, AWS Cloud9, Glitch.com).
```
(Probably the latter.)

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:67]: "GitHub" has a capital "H" and as a brand name should be spelled always the same.
```suggestion
2. Sign up for an account (https://glitch.com/signin) or use your GitHub account if you have one (see GitHub instructions below).
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:74]: For syntax highlighting (not needed here, as this will just be copy-pasted by participants, but looks nicer)
```suggestion
```json
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:90]: For syntax highlighting (not needed here, as this will just be copy-pasted by participants, but looks nicer)
```suggestion
```bash
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:105]: Is there a reason to use triple backticks for inline code here? Default for inline code snippets in Markdown are single backticks:
```suggestion
Once these files are created, open a terminal (in the tools menu at the bottom) and execute the command `refresh`.
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:97]: should probably be
```suggestion
```lang-sh
```
or
```suggestion
```bash
```
but neither make any difference in GitHub's rendering for these code lines. (Whether they'd look different in Honkit's result, I didn't try.)

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:107]: This is a very long sentence. Can it be split up? E.g.
```suggestion
The initial restarting of the Glitch project should fail (complaining that the ```manage.py``` file is missing), but do not worry about it at this stage.
The tutorial will fix this as soon as you create your first Django project.
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:110]: Is there a reason to use triple backticks for inline code here? Default for inline code snippets in Markdown are single backticks:
```suggestion
Simply add the following at the end of the `mysite/settings.py` file.
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:133]: Is there a reason to use triple backticks for inline code here? Default for inline code snippets in Markdown are single backticks:
```suggestion
This will create a `debug.log` file detailing Django operations and any error messages that might come up, making it much easier to fix if your website does not work.
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:67]: If we instruct the participants to sign up, shouldn't the link go to https://glitch.com/signup ?
```suggestion
2. Sign up for an account (https://glitch.com/signup) or use your Github account if you have one (see Github instructions below).
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:68]: This seems to be labeled "hello-webpage" not "Hello Webpage".

Also, on a fresh account, one doesn't have to click on "New Project" (at the upper right) to find that, it's also available at the front-and-center "Let's get started!" section, which might be easier to spot:
glitch.com screenshot with "hello-website" highlighted by a red circle

```suggestion
3. Click _New Project_ and choose _hello-webpage_
```
or simply
```suggestion
3. Click on _hello-webpage_
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:69]: I only see that button (or is it a menu entry?) when I click on "Tools" first.

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:86]: To make it easier to keep this version up to date, please use the existing GitBook variable. Also, prefer `~=` (greatest compatible version) over `>=` (greatest version greater or equal to).
```suggestion
Django~={{ book.django_version }}
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:114]: `mysite/settings.py` hasn't yet been created at this point in the tutorial.

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:135]: If these aren't consecutive lines in `settings.py`, format it as two separate code blocks.

**Review by das-g — CHANGES_REQUESTED** [review]

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:127]: Humans are bad at coming up with truly random stuff. Should we instruct participants to run
```
python -c 'from django.core.management.utils import get_random_secret_key; \
            print(get_random_secret_key())'
```
in the Glitch.com terminal? (command from [here](https://saasitive.com/tutorial/generate-django-secret-key/))

That works in the Glitch.com termianl after the setup instructions added by this pull request.

File: .github/workflows/deploy.yml
Comment (bcopy) [reply .github/workflows/deploy.yml:10]: Also didn't see that one coming, I'll generate from a different branch.

File: en/chromebook_setup/instructions.md
Comment (das-g) [reply en/chromebook_setup/instructions.md:69]: (Also, it looks like the terminal isn't needed until further below, where there's a new (and correct) instruction on how to open it anyway.)

Comment (bcopy) [reply en/chromebook_setup/instructions.md]: I have updated the instructions to match the latest Glitch user interface (Tools -> Terminal)

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md:123]: Indeed, I updated the instructions to run the "first Django project" command first, followed by a glitch `refresh` command, so that the file exists.

File: en/django_start_project/README.md
Comment (bcopy) [reply en/django_start_project/README.md]: Thanks, I created two distinct list items.

File: en/django_start_project/README.md
Comment (bcopy) [reply en/django_start_project/README.md]: Thanks for the tip, I updated and added the instructions to generate the secret key.

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: I added more precise indications to find it.

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Thanks, corrected.

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: I have moved it below the LOGGING settings, and split into two distinct sentences.

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Very true, I corrected it !

**Comment by bcopy** [comment]
Dear @das-g,
Thank you very much for your thorough review, I applied all your recommendations  I believe, let me know what you think.

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:144]: The rest of your contribution has each sentence on a source line of its own, which, although most of the Django Girls tutorial isn't written that way, is IMO a good idea. (It's essentially the most basic form of ["semantic linefeeds"](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) a.k.a. ["semantic line breaks"](https://sembr.org/).) Let's do apply that here, too.
```suggestion
The initial restarting of the Glitch project should fail (if you click on the top dropdown button `Show` then click on `In a New Window`, you will receive a `DisallowedHost` error message).
Do not worry about it at this stage, the tutorial will fix this as soon as you update the Django settings of your project in the `mysite/settings.py` file.
```

Maybe, let's also pull out the parenthesis from the sentence, as it contains a whole full sentence of its own.
```suggestion
The initial restarting of the Glitch project should fail.
(If you click on the top dropdown button `Show` then click on `In a New Window`, you will receive a `DisallowedHost` error message.)
Do not worry about it at this stage, the tutorial will fix this as soon as you update the Django settings of your project in the `mysite/settings.py` file.
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:123]: The rest of your contribution has each sentence on a source line of its own, which, although most of the Django Girls tutorial isn't written that way, is IMO a good idea. (It's essentially the most basic form of ["semantic linefeeds"](https://rhodesmill.org/brandon/2012/one-sentence-per-line/) a.k.a. ["semantic line breaks"](https://sembr.org/).) Let's do apply that here, too.

Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.

```suggestion
>   * First, we are going to create a random secret key.
>     Open the Glitch terminal again, and type the following command:
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:132]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
>   * Create a file `.env` at the root of your project and add the following property in it:
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:138]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
>   * Then update the Django settings file to inject this secret value and set the Django web site name:
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:143]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
>   * And a little further, in the same file, we inject the name of your new Glitch website:
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:208]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
or this one if you are using Glitch:
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:237]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
or on Glitch:
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:76]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
When using Glitch.com as your Cloud IDE, we do not create a virtual environment but we create the following files manually:
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:110]:
```suggestion
Once these files are created, open a terminal (in the tools menu at the bottom) and execute the following commands to create your first Django project:
```

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:122]: Putting a space before a colon (`:`) is AFAIK a French convention not usually followed in English.
```suggestion
> remain confidential (otherwise, anyone remixing your project could see it):
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:67]: Let's pull the parenthesis out of the sentence
```suggestion
2. Sign up for an account (https://glitch.com/signup) or use your GitHub account if you have one. (See GitHub instructions below)
```
or
```suggestion
2. Sign up for an account (https://glitch.com/signup) or use your GitHub account if you have one. (See GitHub instructions below.)
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:76]: The subjects should probably agree within this sentence.
```suggestion
When using Glitch.com as our Cloud IDE, we do not create a virtual environment but we create the following files manually:
```
or
```suggestion
When using Glitch.com as your Cloud IDE, you don't have to create a virtual environment.
Instead, create the following files manually:
```

File: en/chromebook_setup/instructions.md
Comment (das-g) [inline en/chromebook_setup/instructions.md:110]: Isn't the terminal now already open at this point?

File: en/django_start_project/README.md
Comment (das-g) [inline en/django_start_project/README.md:129]:
```suggestion
>     This should display a long random string, perfect to use a as secret key for your brand new Django web site.
```

**Review by das-g — CHANGES_REQUESTED** [review]
Thanks for making the adaptions and changes. Reading the result again, I've noticed some more things, most of them minor:

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Corrected

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Removed backticks

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Applied second suggestion

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Applied

File: en/chromebook_setup/instructions.md
Comment (bcopy) [reply en/chromebook_setup/instructions.md]: Applied

File: en/django_start_project/README.md
Comment (bcopy) [reply en/django_start_project/README.md]: Corrected to : to use as a secret key

**Comment by bcopy** [comment]
Dear @das-g, applied the latest batch of changes, let me know if you can see more corrections.

**Review by das-g — APPROVED** [review]
Looks good to me now.

**Comment by das-g** [comment]
These changes are now already reflected in the English original on Crowdin, so translation can be done there.

**Pull request merged**
